storage_service/locations/forms.py

```diff
@@ -223,6 +226,10 @@ def clean_purpose(self):
             raise django.core.exceptions.ValidationError(_('Invalid purpose'))
         return data
 
+    def save(self, commit=True):
```

Comment:
We override the `save` method so we have a chance to assign the user choice to the current instance of the model.